packages/react/src/components/Icon/Icon.tsx

```diff
@@ -45,50 +67,97 @@ export interface IconProps extends UIComponentProps, ColorComponentProps {
   xSpacing?: IconXSpacing
 }
 
-class Icon extends UIComponent<WithAsProp<IconProps>, any> {
-  static create: ShorthandFactory<IconProps>
+const Icon: React.FC<WithAsProp<IconProps>> & FluentComponentStaticProps = props => {
+  const {
+    accessibility,
+    alt,
+    'aria-label': ariaLabel,
+    bordered,
+    circular,
+    className,
+    color,
+    disabled,
+    design,
+    name,
+    outline,
+    rotate,
+    size,
+    styles,
+    variables,
+    xSpacing,
+  } = props
+
+  const context: ProviderContextPrepared = React.useContext(ThemeContext)
+  const { setStart, setEnd } = useTelemetry(Icon.displayName, context.telemetry)
+
+  setStart()
+
+  const getA11Props = useAccessibility(accessibility, {
+    debugName: Icon.displayName,
+    mapPropsToBehavior: () => ({
+      alt,
+      'aria-label': ariaLabel,
+    }),
+    rtl: context.rtl,
+  })
+  const { classes } = useStyles(Icon.displayName, {
+    className: Icon.className,
+    mapPropsToStyles: () => ({
+      bordered,
+      circular,
+      color,
+      disabled,
+      name,
+      outline,
+      rotate,
+      size,
+      xSpacing,
+    }),
+    mapPropsToInlineStyles: () => ({ className, design, styles, variables }),
+    rtl: context.rtl,
+  })
 
-  static className = 'ui-icon'
+  const ElementType = getElementType(props)
+  const unhandledProps = getUnhandledProps(Icon.handledProps, props)
 
-  static displayName = 'Icon'
+  const { icons = {} } = context.theme
+  const maybeIcon = icons[name]
+  const isSvgIcon = maybeIcon && maybeIcon.isSvg
 
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      children: false,
-      content: false,
-      color: true,
-    }),
-    bordered: PropTypes.bool,
-    circular: PropTypes.bool,
-    disabled: PropTypes.bool,
-    name: PropTypes.string.isRequired,
-    outline: PropTypes.bool,
-    rotate: PropTypes.number,
-    size: customPropTypes.size,
-    xSpacing: PropTypes.oneOf(['none', 'before', 'after', 'both']),
-  }
-
-  static defaultProps = {
-    as: 'span',
-    size: 'medium',
-    accessibility: iconBehavior,
-    rotate: 0,
-  }
-
-  renderComponent({ ElementType, classes, unhandledProps, accessibility, theme, rtl, styles }) {
-    const { name } = this.props
-    const { icons = {} } = theme || {}
-
-    const maybeIcon = icons[name]
-    const isSvgIcon = maybeIcon && maybeIcon.isSvg
-
-    return (
-      <ElementType className={classes.root} {...accessibility.attributes.root} {...unhandledProps}>
-        {isSvgIcon && callable(maybeIcon.icon)({ classes, rtl, props: this.props })}
-      </ElementType>
-    )
-  }
+  setEnd()
+
+  return (
+    <ElementType {...getA11Props('root', { className: classes.root, ...unhandledProps })}>
+      {isSvgIcon && callable(maybeIcon.icon)({ classes, rtl: context.rtl, props })}
```

Comment:
Maybe we should restrict the props sent here too. Not as part of this PR, though

Reply:
Agree, let's do it separately. I am even not sure that should pass there something